kafka-connect-aws-s3/src/main/scala/io/lenses/streamreactor/connect/aws/s3/sink/config/PartitionFieldSplitter.scala

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2017-2023 Lenses.io Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.lenses.streamreactor.connect.aws.s3.sink.config
+
+/**
+  * A utility for splitting a string into parts while respecting backticks.
+  */
+object PartitionFieldSplitter {
+
+  /**
+    * Splits an input string into parts while respecting backticks.
+    * Backticks are treated as complete strings and not subject to splitting.
+    *
+    * @param input The input string to split.
+    * @return A sequence of strings representing the split parts.
+    */
+  def split(input: String): Seq[String] =
```

Comment:
```scala
    val input = "_value.field.`a.b`.c.d.`e.f.g`"
    val pattern  = """`([^`]+)`|([^.`]+)""".r

    
    val result = pattern.findAllMatchIn(input).map { matchResult =>
      Option(matchResult.group(1)).getOrElse(matchResult.group(2))
    }.toSeq
 
   result.foreach(println)
```

Achieves the same outcome. might be a bit easier to read (apart from the regex :) )